apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+pragma solidity 0.4.15;
+
+import "./oracle/OracleInterface.sol";
+
+import "@aragon/core/contracts/apps/App.sol";
+import "@aragon/core/contracts/common/Initializable.sol";
+import "@aragon/core/contracts/common/EtherToken.sol";
+import "@aragon/core/contracts/common/erc677/ERC677Receiver.sol";
+
+import "@aragon/core/contracts/zeppelin/token/ERC20.sol";
+import "@aragon/core/contracts/zeppelin/math/SafeMath.sol";
+
+import "@aragon/apps-finance/contracts/Finance.sol";
+//import "@aragon/apps-vault/contracts/Vault.sol";
+
+
+/**
+ * @title Payroll in multiple currencies
+ * @notice For the sake of simplicity lets asume USD is a ERC20 token
+ * Also lets asume we can 100% trust the exchange rate oracle
+ */
+contract Payroll is App, Initializable, ERC677Receiver {
+    using SafeMath for uint;
+
+    bytes32 constant public OWNER_WRITE_ROLE = bytes32(1);
+    bytes32 constant public OWNER_READ_ROLE = bytes32(2);
+    bytes32 constant public EMPLOYEE_ROLE = bytes32(3);
+    bytes32 constant public ORACLE_ROLE = bytes32(4);
```

Comment:
Same with oracle role but in reverse, you could remove keeping track of the oracle in this contract and just have an oracle role that is managed in the kernel.

Reply:
Well, oracle is used in checkExchangeRate. Although it overcomplicates contract a bit, the idea was to allow both directions: if oracle hasn't set an exchange rate, we can try to query it. The final goal was to try to save `payday` transactions and avoid some frustration for the user(employee) not being able to have the salary paid because of a missing exchange rate. Maybe it's unnecessary.

Reply:
Fair enough. I would remove the complexity though